### PR TITLE
feat(warehouse): added additional datatypes for warehouse integrations

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

- Additional datatypes for warehouse destination. These are collected from `warehouse_rudder_missing_datatype` stats generated during the fetching of the schema from the warehouse.
- Some datatypes got missed as part of the alias supported by the warehouses.

### Corresponding documentation links:

1. [BQ](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types)
2. [CLICKHOUSE](https://clickhouse.com/docs/en/sql-reference/data-types/)
3. [DELTALAKE](https://docs.databricks.com/sql/language-manual/sql-ref-datatypes.html#exact-numeric), [ODBC](https://dev.mysql.com/doc/connector-odbc/en/connector-odbc-reference-datatypes.html)
4. [MSSQL](https://learn.microsoft.com/en-us/sql/t-sql/data-types/data-types-transact-sql?view=sql-server-ver16)
5. [POSTGRES](https://www.postgresql.org/docs/current/datatype.html), [bytea](https://www.postgresql.org/docs/current/datatype-binary.html)
6. [RS](https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html)
7. [S3_DATALAKE](https://docs.trifacta.com/display/r071/AWS+Glue+Data+Type+Conversions)
8. [SNOWFLAKE](https://docs.snowflake.com/en/sql-reference/data-types.html)

## Notion Ticket

https://www.notion.so/rudderstacks/Additional-datatypes-for-warehouse-Integrations-a765dda055db40589f0b2eedddce56ad?pvs=4

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by lvrach — APPROVED** [review]

**Comment by achettyiitr** [comment]
https://github.com/rudderlabs/rudder-server/pull/2636/

**Comment by achettyiitr** [comment]
DeepSource issues are resolved in this https://github.com/rudderlabs/rudder-server/pull/2964

File: warehouse/integrations/mssql/mssql.go [hunk 1]
```diff
@@ -57,28 +57,43 @@ var rudderDataTypesMapToMssql = map[string]string{
...
+	"datetime":                 "datetime",
+	"time":                     "datetime",
 	"timestamp with time zone": "datetime",
 	"timestamp":                "datetime",
-	"jsonb":                    "json",
-	"bit":                      "boolean",
+
+	"bit": "boolean",
+
+	"jsonb": "json",
```
Comment (achettyiitr) [inline warehouse/integrations/mssql/mssql.go:96]: These datatypes we have added: 
int, money, small money, binary, image, varbinary, uniqueidentifier

File: warehouse/integrations/bigquery/bigquery.go [hunk 2]
```diff
@@ -70,15 +70,28 @@ var dataTypesMap = map[string]bigquery.FieldType{
...
+	"DECIMAL":    "float",
+	"BIGNUMERIC": "float",
+	"BIGDECIMAL": "float",
+	"FLOAT":      "float",
+	"FLOAT64":    "float",
+
+	"STRING": "string",
+	"JSON":   "string",
+	"BYTES":  "string",
+
```
Comment (achettyiitr) [inline warehouse/integrations/bigquery/bigquery.go:94]: These datatypes we have added.
INT, SMALLINT, BIGINT, TINYINT, BYTEINT, DECIMAL, BIGNUMERIC, BIGDECIMAL

File: warehouse/integrations/clickhouse/clickhouse.go [hunk 3]
```diff
@@ -85,36 +85,78 @@ var datatypeDefaultValuesMap = map[string]interface{}{
...
+	"Nullable(Int16)":           "int",
+	"Nullable(Int32)":           "int",
+	"Nullable(Int64)":           "int",
+	"Nullable(Float32)":         "float",
+	"Nullable(Float64)":         "float",
+	"Nullable(String)":          "string",
+	"Nullable(UUID)":            "string",
+	"Nullable(DateTime)":        "datetime",
+	"Nullable(DateTime('UTC'))": "datetime",
+	"Nullable(UInt8)":           "boolean",
```
Comment (achettyiitr) [inline warehouse/integrations/clickhouse/clickhouse.go:158]: We have added these datatypes.
TINYINT,  INT1, SMALLINT, INT2, INT, INT4, INTEGER, BIGINT, INT128, INT256, FLOAT, DOUBLE, LONGTEXT, MEDIUMTEXT, TINYTEXT, TEXT, LONGBLOB, MEDIUMBLOB, TINYBLOB, BLOB, VARCHAR, CHAR, UUID, DateTime('UTC'), DateTime64, DateTime64('UTC'), Nullable(UUID), Nullable(DateTime('UTC'))

**Pull request closed without merging**